src/Microsoft.SqlTools.ServiceLayer/ObjectExplorer/SmoModel/DatabaseTreeNode.cs

```diff
@@ -102,5 +104,12 @@ private bool IsDWGen3(Database db)
                 && db.DatabaseEngineEdition == DatabaseEngineEdition.SqlDataWarehouse
                 && db.ServerVersion.Major == 12;
         }
+
+        private bool IsDataverse(Database db){
+            // Dataverses have a DatabaseEngineEdition of 1000. We have to cast it to int as DatabaseEngineEdition enum does not have a value for dataverse.
+            return db!= null
+                && (int)db.DatabaseEngineEdition == 1000 
+                && db.ServerVersion.Major == 12;
```

Comment:
> 12; [](http://example.com/codeflow?start=44&length=4)

they could upgrade their version number at any time. Does it really matter what version they report?